datapipelines/src/main/kotlin/io/customer/sdk/CustomerIO.kt

```diff
@@ -310,14 +323,18 @@ class CustomerIO private constructor(
             return
         }
 
-        track(EventNames.DEVICE_DELETE)
+        trackDelete(enrichmentClosure = enrichment)
+    }
+
+    private fun trackDelete(enrichmentClosure: EnrichmentClosure?) {
+        track(name = EventNames.DEVICE_DELETE, properties = emptyJsonObject, serializationStrategy = JsonAnySerializer.serializersModule.serializer(), enrichment = enrichmentClosure)
```

Comment:
I see `trackDelete` only called once, so either this can be replaced with code directly, or should be more generic, something like following

```kotlin
private fun trackWithEmptyProperties(name:String, enrichmentClosure: EnrichmentClosure?) {
        track(name = name, properties = emptyJsonObject, serializationStrategy = JsonAnySerializer.serializersModule.serializer(), enrichment = enrichmentClosure)
}
```

Function name can surely be improved though